components/concept/storage/src/main/java/mozilla/components/concept/storage/HistoryStorage.kt

```diff
@@ -14,7 +14,7 @@ interface HistoryStorage : Storage {
      * @param uri of the page which was visited.
      * @param visitType type of the visit, see [VisitType].
      */
-    suspend fun recordVisit(uri: String, visitType: VisitType)
+    suspend fun recordVisit(uri: String, visit: PageVisit)
```

Comment:
nit: I like adding the 'RedirectSource' enum.  But I'm  not sure with the decision of adding a new data class.  Looks like we can get away with just adding a extra parameter for recordVisit().  I think just adding a extra parameter makes it easier to read.

Reply:
I thought about adding the source as an extra parameter, but decided against it because we might also want to [track](https://bugzilla.mozilla.org/show_bug.cgi?id=468710) and eventually [sync](https://bugzilla.mozilla.org/show_bug.cgi?id=623667) referrer chains, which we don't currently do. Four params might get unwieldy—but maybe it's premature to worry about that now.

Reply:
I see. if there's a reason then I'm good with it.